### PR TITLE
Parent updated ui

### PR DESCRIPTION
**Pull request by johannkm**
Add a details modal for which parent was updated

unpartitioned
Screenshot 2023-07-20 at 11 21 55 AM
Screenshot 2023-07-20 at 11 21 58 AM

partitioned
Screenshot 2023-07-20 at 10 05 15 PM

### CODE REVIEW

**Comment by johannkm** [comment]
Current dependencies on/for this PR:
* master
  * **PR #15200** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/15200" target="_blank">Graphite</a> 
    * **PR #15204** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/15204" target="_blank">Graphite</a> 
      * **PR #15432** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/15432" target="_blank">Graphite</a>  👈
      * **PR #15401** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/15401" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/15432?utm_source=stack-comment).

**Comment by johannkm** [comment]
A nit, it summarizes 'updated' and 'will update' as 'updated'. Instead it could just say `1 parent asset` which I think would make sense given the context of the condition?

We also could explain `will update` more, with `will update in the same run`. This might be confusing when the parent is across code locations/partitions. `will update in this evaluation`?

**Comment by johannkm** [comment]
@hellendag up to you if you think we should make some of this in to a generic component to share with the waiting on dialog. No more special details are planned right now but they'll come eventually. If so maybe you could commandeer or do a follow on diff

File: js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/ParentUpdatedLink.tsx [hunk 1]
```diff
@@ -0,0 +1,150 @@
+import {
```
Comment (hellendag) [inline js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/ParentUpdatedLink.tsx:1]: Are these basically the same as the existing dialogs?

**Review by hellendag — COMMENTED** [review]
It would definitely be nice to avoid duplicating too much. Are there currently any differences between the existing link/dialog components and these?

File: js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/ParentUpdatedLink.tsx (hunk 1 above)
Comment (johannkm) [reply js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/ParentUpdatedLink.tsx:1]: The new dialog distinguishes 2 lists of asset keys- updated assets and willUpdate asssets

Comment (johannkm) [reply js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/ParentUpdatedLink.tsx:1]: @hellendag I don't have a strong opinion on if we should make the modal a generic component, but doing so would take me a while. Could we
- land this so we can surface the details now, and punt on making it generic
- or have you commandeer and have your way with it?

File: js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/VirtualizedAssetListForDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+              >
+                {renderItem(assetKey)}
+              </Box>
+            </Row>
+          );
+        })}
+      </Inner>
+    </Container>
+  );
+}
```
Comment (hellendag) [inline js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/VirtualizedAssetListForDialog.tsx:49]: Generic virtualized list for showing a list of assets in one of these dialogs.

File: js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/VirtualizedAssetPartitionListForDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+  height: 100%;
+  width: 100%;
+  flex-direction: row;
+  align-items: center;
+  gap: 8px;
+
+  :focus {
+    outline: none;
+  }
+`;
```
Comment (hellendag) [inline js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/VirtualizedAssetPartitionListForDialog.tsx:146]: Generic virtualized list for showing a list of partitioned assets (grouped by partition name, collapsible, etc) in one of these dialogs.

**Comment by hellendag** [comment]
Chatting with @johannkm on Slack -- I pushed up some changes to try to make some of the pieces here more generic and reusable. Things look good to me locally, but I'll let you guys poke at it a bit more to make sure I didn't overlook anything.

**Comment by johannkm** [comment]
I've played around with it, LGTM

**Review by hellendag — APPROVED** [review]

**Pull request merged**